src/server/upgrade.rs

```diff
@@ -138,17 +138,20 @@ pub fn upgrade(options: &Upgrade) -> anyhow::Result<()> {
     let os = detect::current_os()?;
     let methods = os.get_available_methods()?.instantiate_all(&*os, true)?;
     let mut errors = Vec::new();
+    let mut any_upgraded = false;
     for meth in methods.values() {
         match meth.upgrade(&todo, options) {
-            Ok(_) => {
+            Ok(upgraded) => {
+                if upgraded {
+                    any_upgraded = true;
+                }
                 if let ToDo::InstanceUpgrade(name, _version) = &todo {
                     let new_inst = meth.get_instance(name)?;
-                    println!();
-                    print_markdown!(
-                        "**EdgeDB instance '${name}' was successfully \
-                        upgraded to version ${version}!**",
-                        name=name,
-                        version=new_inst.get_current_version()?.unwrap(),
+                    let version = new_inst.get_current_version()?.unwrap();
+                    print::success_msg(
+                        &format!("Successfully upgraded EdgeDB instance \
+                                      '{}' to version", name),
+                        &format!("{}", version)
```

Comment:
I like this way better than abusing markdown for simple status messages. I think there were a few more places with newly added `print_markdown()` calls for that in `master`. Can you rebase and take a look?

Reply:
I'm not sure I like it more. Markdown gives more flexibility, and doesn't require tons of similar functions like `success` vs `success_msg`. The problem with markdown is that it has limited set of styles, unfortunately, so it's not great either :(

Reply:
> The problem with markdown is that it has limited set of styles

This, exactly.

I'm not against using markdown, I just think we should use it wisely. IMO having a function to print successful messages in green color is better than using `print_markdown!("# success")`, at the very least for better readability of the code. But up to you, I'm not trying to impose my opinion here too much.